oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/server/WebSessionOAuth2ServerAuthorizationRequestRepositoryTests.java

```diff
@@ -99,8 +103,175 @@ public void loadAuthorizationRequestWhenSavedThenAuthorizationRequest() {
 				.verifyComplete();
 	}
 
+	/**
+	 * Redis and Hazelcast distributed session implementation only update the change to underlay storage.
+	 * This is a simulation of such distributed session implementation.
+	 */
+	private static class DeltaOptimizedWebSession extends MockWebSession {
```

Comment:
I think we should be able to use a Mock object here instead of extending MockWebSession

Reply:
Let me see how to use a Mock object here.